homeassistant/components/amazon_devices/diagnostics.py

```diff
@@ -0,0 +1,49 @@
+"""Diagnostics support for Amazon Devices integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .coordinator import AmazonConfigEntry
+
+TO_REDACT = {CONF_PASSWORD, CONF_USERNAME, CONF_NAME, "title"}
+
+
+async def async_get_config_entry_diagnostics(
```

Comment:
We can also do device diagnostics, so users can get it from the device page and get ones specific to that device

Reply:
Done!